File: src/main/java/ma/fstg/gestionprojets/services/ProfilService.java
package ma.fstg.gestionprojets.services;
import ma.fstg.gestionprojets.dto.request.ProfilRequest;
import ma.fstg.gestionprojets.dto.response.ProfilResponse;
import java.util.List;
public interface ProfilService {
    ProfilResponse create(ProfilRequest req);
    ProfilResponse update(Long id, ProfilRequest req);
    ProfilResponse getById(Long id);
    List<ProfilResponse> getAll();
    void delete(Long id);
}

File: gestion-projets-frontend/src/services/profilService.ts
import api from '../utils/axiosConfig';

export const profilService = {
  getAll: () => api.get('/profils'),
  getById: (id: number) => api.get(`/profils/${id}`),
  create: (data: any) => api.post('/profils', data),
  update: (id: number, data: any) => api.put(`/profils/${id}`, data),
  delete: (id: number) => api.delete(`/profils/${id}`),
};
